src/osemgrep/cli_interactive/Interactive_subcommand.ml

```diff
@@ -398,45 +431,64 @@ let matches_of_new_iformula (new_iform : iformula) (state : state) :
       filter_irrelevant_rules = true;
     }
   in
-  let res : Report.rule_profiling Report.match_result list =
+  if !(state.should_continue_iterating_targets) then
     state.xtargets
-    |> Common.map_filter (fun xtarget_prot ->
+    (* First, sort files. We will search them in alphabetical order. *)
+    |> List.sort (fun x1 x2 ->
+           Lock_protected.with_lock
+             (fun x1 ->
+               Lock_protected.with_lock
+                 (fun x2 -> Fpath.compare x1.Xtarget.file x2.Xtarget.file)
+                 x2)
+             x1)
+    |> List.iter (fun xtarget_prot ->
            if !(state.should_continue_iterating_targets) then
              xtarget_prot
              |> Lock_protected.with_lock (fun xtarget ->
-                    if
-                      Match_rules.is_relevant_rule_for_xtarget fake_rule xconf
-                        xtarget
-                    then
-                      (* Calling the engine! *)
-                      Some
-                        (Match_search_mode.check_rule fake_rule hook xconf
-                           xtarget)
-                    else None)
+                    let+ { matches; _ } =
+                      if
+                        Match_rules.is_relevant_rule_for_xtarget fake_rule xconf
+                          xtarget
+                      then
+                        (* Calling the engine! *)
+                        Some
+                          (Match_search_mode.check_rule fake_rule hook xconf
+                             xtarget)
+                      else None
+                    in
+                    matches
```

Comment:
this function is getting pretty long. Maybe you can introduce an helper function with a useful name.